code/go/0chain.net/smartcontract/storagesc/blobber.go

```diff
@@ -72,102 +74,207 @@ func (sc *StorageSmartContract) hasBlobberUrl(blobberURL string,
 	}
 }
 
+func validateBlobberUpdateSettings(updateBlobberRequest *dto.StorageDtoNode, conf *Config) error {
+	if updateBlobberRequest.Capacity != nil && *updateBlobberRequest.Capacity <= conf.MinBlobberCapacity {
+		return errors.New("insufficient blobber capacity in update blobber settings")
+	}
+
+	if err := validateBaseUrl(updateBlobberRequest.BaseURL); err != nil {
+		return err
+	}
+
+	return nil
+}
+
 // update existing blobber, or reborn a deleted one
 func (sc *StorageSmartContract) updateBlobber(
-	t *transaction.Transaction,
+	txn *transaction.Transaction,
 	conf *Config,
-	blobber, savedBlobber *StorageNode,
-	sp *stakePool,
+	updateBlobberRequest *dto.StorageDtoNode,
+	existingBlobber *StorageNode,
+	existingStakePool *stakePool,
 	balances cstate.StateContextI,
 ) (err error) {
-	// check terms
-	if err = blobber.Terms.validate(conf); err != nil {
-		return fmt.Errorf("invalid blobber terms: %v", err)
+	// validate the new terms and update the existing blobber's terms
+	if err = validateTermsInUpdatedRequestAndMergeInSavedBlobber(updateBlobberRequest, existingBlobber, conf); err != nil {
+		return err
 	}
 
-	if blobber.Capacity <= 0 {
-		return sc.removeBlobber(t, blobber, balances)
+	if err = validateGeolocationInUpdatedRequestAndMergeInSavedBlobber(updateBlobberRequest, existingBlobber); err != nil {
```

Comment:
same here, just a short quote from effective go

```
Another short example is once.Do; once.Do(setup) reads well and would not be improved by writing once.DoOrWaitUntilDone(setup). Long names don't automatically make things more readable. A helpful doc comment can often be more valuable than an extra long name.
```

Reply:
Renaming to validateAndSaveGeoLoc() and the terms func to validateAndSaveTerms()